subsys/zigbee/cli/zigbee_cli_cmd_generic_cmd.c

```diff
@@ -362,6 +363,12 @@ int cmd_zb_generic_cmd(const struct shell *shell, size_t argc, char **argv)
 
 	/* Check if payload should be sent. */
 	if (strcmp(*(arg++), "-l") == 0) {
+		/* In case payload starts with 0x, skip it and warn user. */
+		if (((*arg)[0] == '0') && (tolower((*arg)[1]) == 'x')) {
+			*arg += 2;
+			shell_warn(shell, "Trimming \"0x\" from the payload");
```

Comment:
Do we need this warning or this is an expected behavior?

Reply:
Providing payload that starts with `0x` may suggest that user treats the payload as one big number, which is not correct for the most part, also the endianess.

Reply:
We can skip `0x` silently without warning user, sure. Just not sure if this is the best idea

Reply:
@robertstypa Since you had tried to do such thing, what would your preferred behavior?

Reply:
@tomchy I discussed that with @robertstypa, we agreed on skipping the `0x` and warning user about that.
- Skip `0x` to fix isue with additional byte being sent
- Warn user about that trimming because `0x` in payload is not especially correct, as the payload can be constructed from multiple values of different type and length.

Reply:
Than I'm fine with this implementation.
The `0x` may have also the meaning `I have to write "A", but I want to make sure that it will not be interpreted as ASCII`.